boofuzz/fuzz_logger.py

```diff
@@ -87,3 +87,6 @@ def failure_summary(self):
             summary += "{0}".format('\n'.join(map(str, self.error_test_cases.iterkeys())))
 
         return summary
+
+    def write_log(self, force=False):
+        self._fuzz_loggers[0].write_log(force=force)
```

Comment:
From reading the code, it looks like this method is used to signal the end of a test case (or the end of an entire test). One concern I had initially is that the implementation details of the fuzz logger (e.g., when data should be written) were being leaked outside the class. However, once I realized that it essentially signals completion of a test case, it made more sense. Here are some readability/usability suggestions:

1. Rename the method to `end_test_case`.
2. Add an `end_test` method.
3. Add these methods to ifuzz_logger.py, and to each of the fuzz logger implementation (text, csv) for consistency. They can do nothing or log a simple message in those cases.
4. Within `fuzz_logger_db.py`, rename `write_log` to `_write_log`. The `end_test_case` method will call `_write_log` without force. The `end_test` method will call it with force.

This allows us to keep all content regarding database optimization strictly within the `FuzzLoggerDb` class. It will also make the code within `FuzzLoggerDb` more readable to a newcomer, as the public methods will make clear how and when `_write_log` is called.

Reply:
That should make the code more readable, will rework.